client-js/app/elements/app-link.ts

```diff
@@ -0,0 +1,30 @@
+const MOUSE_MAIN_BUTTON = 0;
+
+@component('app-link')
+@extend('a')
+export class AppLink extends polymer.Base {
+
+  @property({type: Array})
+  path: string[];
+
+  isSpecialClick(e): boolean {
+    return (e.ctrlKey || e.metaKey || e.button !== MOUSE_MAIN_BUTTON);
+  }
+
+  @listen('click')
+  onClick(e) {
+    if (this.isSpecialClick(e)) return;
+
+    this.fire('app-link-click', {path: this.path});
```

Comment:
For js at least it's the former: https://google.github.io/styleguide/javascriptguide.xml?showone=Code_formatting#Code_formatting